env_support/cmake/custom.cmake

```diff
@@ -46,37 +46,54 @@ if(LV_CONF_SKIP)
   target_compile_definitions(lvgl PUBLIC LV_CONF_SKIP=1)
 endif()
 
+
+
+
 # Include root and optional parent path of LV_CONF_PATH
 target_include_directories(lvgl SYSTEM PUBLIC ${LVGL_ROOT_DIR} ${LV_CONF_DIR} ${CMAKE_CURRENT_BINARY_DIR})
 
+if(LV_CONF_BUILD_DISABLE_THORVG_INTERNAL)
+  message(FATAL_ERROR "LV_CONF_BUILD_DISABLE_THORVG_INTERNAL is deprecated, use CONFIG_LV_USE_THORVG_INTERNAL instead")
+endif()
 
-if(NOT LV_CONF_BUILD_DISABLE_THORVG_INTERNAL)
+if(LV_CONF_BUILD_THORVG_INTERNAL)
     add_library(lvgl_thorvg ${THORVG_SOURCES})
     add_library(lvgl::thorvg ALIAS lvgl_thorvg)
     target_include_directories(lvgl_thorvg SYSTEM PUBLIC ${LVGL_ROOT_DIR}/src/libs/thorvg)
     target_link_libraries(lvgl_thorvg PUBLIC lvgl)
+    target_compile_definitions(lvgl_thorvg PUBLIC LV_USE_THORVG_INTERNAL=1 LV_USE_THORVG_EXTERNAL=0)
 endif()
 
-if(NOT (CMAKE_C_COMPILER_ID STREQUAL "MSVC"))
+if(NOT MSVC)
   set_source_files_properties(${LVGL_ROOT_DIR}/src/others/vg_lite_tvg/vg_lite_tvg.cpp PROPERTIES COMPILE_FLAGS -Wunused-parameter)
 endif()
 
+if(LV_CONF_BUILD_DISABLE_EXAMPLES)
+  message(FATAL_ERROR "LV_CONF_BUILD_DISABLE_EXAMPLES is deprecated, use CONFIG_LV_BUILD_EXAMPLES instead")
+endif()
+
 # Build LVGL example library
-if(NOT LV_CONF_BUILD_DISABLE_EXAMPLES)
+if(CONFIG_LV_BUILD_EXAMPLES)
     add_library(lvgl_examples ${EXAMPLE_SOURCES})
     add_library(lvgl::examples ALIAS lvgl_examples)
 
     target_include_directories(lvgl_examples SYSTEM PUBLIC ${LVGL_ROOT_DIR}/examples)
     target_link_libraries(lvgl_examples PUBLIC lvgl)
+    target_compile_definitions(lvgl_examples PUBLIC LV_BUILD_EXAMPLES=1)
+endif()
+
+if(LV_CONF_BUILD_DISABLE_DEMOS)
+  message(FATAL_ERROR "LV_CONF_BUILD_DISABLE_DEMOS is deprecated, use CONFIG_LV_USE_DEMO_WIDGETS instead")
 endif()
 
 # Build LVGL demos library
-if(NOT LV_CONF_BUILD_DISABLE_DEMOS)
+if(CONFIG_LV_USE_DEMO_WIDGETS)
     add_library(lvgl_demos ${DEMO_SOURCES})
     add_library(lvgl::demos ALIAS lvgl_demos)
 
     target_include_directories(lvgl_demos SYSTEM PUBLIC ${LVGL_ROOT_DIR}/demos)
     target_link_libraries(lvgl_demos PUBLIC lvgl)
+    target_compile_definitions(lvgl_demos PUBLIC LV_USE_DEMO_WIDGETS=1)
```

Comment:
Unrelated to the CI, I think `CONFIG_LV_USE_DEMO_WIDGETS` should be `CONFIG_LV_BUILD_DEMOS`, and maybe we shouldn't enable the widgets demo in here? `DEMO_WIDGETS` is just one of the `DEMOS`.

Reply:
allright replaced CONFIG_LV_USE_DEMO_WIDGETS with CONFIG_LV_BUILD_DEMOS.
If we would like to get rid of CONFIG_LV_BUILD_DEMOS we should remove the define `LV_USE_DEMO_WIDGETS=1`. But lets handle this in a separate PR. This PR is getting far too big :)